### PR TITLE
Add api framework

### PR DESCRIPTION
**Pull request by BetaDraconis**

### CODE REVIEW

**Comment by BetaDraconis** [comment]
Test comment

File: README.md [hunk 1]
```diff
@@ -1,8 +1,42 @@
+insolvent-vat-api
+========================
```
Comment (BetaDraconis) [inline README.md:2]: test review comment

File: app/config/AppConfig.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  def apiGatewayContext: String
+
+  def apiStatus(version: String): String
+  def featureSwitch: Option[Configuration]
+  def endpointsEnabled(version: String): Boolean
+}
+
+@Singleton
+class AppConfigImpl @Inject()(config: ServicesConfig, configuration: Configuration) extends AppConfig {
+  val mtdIdBaseUrl: String = config.baseUrl("mtd-id-lookup")
```
Comment (narendravijayarao) [inline app/config/AppConfig.scala]: do we need **mtd-id-lookup** config?

File: app/config/DocumentationController.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+
+@Singleton
+class DocumentationController @Inject()(selfAssessmentApiDefinition: ApiDefinitionFactory,
+                                        cc: ControllerComponents, assets: Assets, errorHandler: HttpErrorHandler)
+  extends HmrcDocumentationController(cc,assets , errorHandler ) {
+
+  override def definition(): Action[AnyContent] = Action {
+    Ok(Json.toJson(selfAssessmentApiDefinition.definition))
```
Comment (narendravijayarao) [inline app/config/DocumentationController.scala]: selfAssessmentApiDefinition?

File: app/v1/connectors/MtdIdLookupConnector.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
```
Comment (narendravijayarao) [inline app/v1/connectors/MtdIdLookupConnector.scala]: Do we need this class??

File: app/v1/connectors/httpparsers/MtdIdLookupHttpParser.scala [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (narendravijayarao) [inline app/v1/connectors/httpparsers/MtdIdLookupHttpParser.scala]: Can delete this class

**Pull request merged**